src/main/java/com/synopsys/integration/alert/channel/slack/SlackChannel.java

```diff
@@ -87,8 +86,23 @@ public String sendMessage(final DistributionEvent event) throws IntegrationExcep
 
     public List<Request> createRequests(final DistributionEvent event) throws IntegrationException {
         final FieldAccessor fields = event.getFieldAccessor();
-        final String webhook = fields.getString(SlackDescriptor.KEY_WEBHOOK).orElseThrow(() -> new AlertException("Missing Webhook URL"));
-        final String channelName = fields.getString(SlackDescriptor.KEY_CHANNEL_NAME).orElseThrow(() -> new AlertException("Missing channel name"));
+
+        final Optional<String> webhookOptional = fields.getString(SlackDescriptor.KEY_WEBHOOK);
+        final Optional<String> channelNameOptional = fields.getString(SlackDescriptor.KEY_CHANNEL_NAME);
+
+        if (webhookOptional.isEmpty() || channelNameOptional.isEmpty()) {
+            Map<String, String> fieldErrors = new HashMap();
+            if (webhookOptional.isEmpty()) {
+                fieldErrors.put(SlackDescriptor.KEY_WEBHOOK, "Missing Webhook URL");
+            }
+            if (channelNameOptional.isEmpty()) {
+                fieldErrors.put(SlackDescriptor.KEY_CHANNEL_NAME, "Missing channel name");
+            }
+            throw new AlertFieldException(fieldErrors);
+        }
+        String webhook = webhookOptional.get();
```

Comment:
I really prefer using `.isPresent()` with `.get()` that way it is nested in the code-block.

Reply:
So you would prefer something like this?
```
String webhook = "";
        String channelName = "";

        if (webhookOptional.isPresent() && channelNameOptional.isPresent()) {
            webhook = webhookOptional.get();
            channelName = channelNameOptional.get();
        } else {
            Map<String, String> fieldErrors = new HashMap();
            if (webhookOptional.isEmpty()) {
                fieldErrors.put(SlackDescriptor.KEY_WEBHOOK, "Missing Webhook URL");
            }
            if (channelNameOptional.isEmpty()) {
                fieldErrors.put(SlackDescriptor.KEY_CHANNEL_NAME, "Missing channel name");
            }
            throw new AlertFieldException(fieldErrors);
        }
```

Reply:
I would prefer all of the requiring the optional fields to be inside of the if block.